src/data/matrix/hadamard.lean

```diff
@@ -0,0 +1,140 @@
+/-
+Copyright (c) 2021 Lu-Ming Zhang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Lu-Ming Zhang
+-/
+import data.matrix.notation
+import linear_algebra.matrix.trace
+import data.complex.basic
+
+/-!
+# Hadamard product of matrices
+
+This file defines the Hadamard product `matrix.Hadamard`
+and contains basic properties about them.
+
+## Main definition
+
+- `matrix.Hadamard`: defines the Hadamard product,
+  which is the pointwise product of two matrices of the same size.
+
+## Notation
+
+* `⊙`: the Hadamard product `matrix.Hadamard`;
+
+## References
+
+*  <https://en.wikipedia.org/wiki/Hadamard_product_(matrices)>
+
+## Tags
+
+Hadamard product, Kronecker product, Hadamard, Kronecker
+-/
+
+variables {α β γ I J K L M N: Type*}
+variables {R : Type*}
+variables {m n p q r s t: ℕ}
+variables [fintype I] [fintype J] [fintype K] [fintype L] [fintype M] [fintype N]
+
+namespace matrix
+open_locale matrix big_operators
+open complex
+
+def Hadamard [has_mul α] (A : matrix I J α) (B : matrix I J α) :
+matrix I J α :=
+λ i j, (A i j) * (B i j)
```

Comment:
A slightly wacky idea
```suggestion
@has_mul.mul (matrix I J α) pi.has_mul A B
```
Which would make all the proofs below even more trivial

Reply:
I think the two definitions (Eric's and the one you have right now) are defeq?

```
example [has_mul α] (A B : matrix I J α) : Hadamard A B = @has_mul.mul _ pi.has_mul A B := rfl
```